pandas/_libs/algos_common_helper.pxi.in

```diff
@@ -4,32 +4,34 @@ Template for each `dtype` helper function using 1-d template
 WARNING: DO NOT edit .pxi FILE directly, .pxi is generated from .pxi.in
 """
 
-{{py:
-
-# name, c_type, dest_type
-dtypes = [('float64', 'float64_t', 'float64_t'),
-          ('float32', 'float32_t', 'float32_t'),
-          ('int8',  'int8_t',  'float32_t'),
-          ('int16', 'int16_t', 'float32_t'),
-          ('int32', 'int32_t', 'float64_t'),
-          ('int64', 'int64_t', 'float64_t')]
-
-def get_dispatch(dtypes):
-
-    for name, c_type, dest_type, in dtypes:
-        yield name, c_type, dest_type
+ctypedef fused diff_t:
```

Comment:
isn’t this numeric? can we consolidate these later

Reply:
i hope so